device_coupler/ovs_helper.py

```diff
@@ -80,18 +87,28 @@ def set_trunk_vlan(self, interface, vlans):
         vlan_str = ",".join(str(vlan) for vlan in vlans)
         self._run_shell('sudo ovs-vsctl set port %s trunks=%s' % (interface, vlan_str))
 
-    def create_faux_device(self, index):
-        """Creates faux docker container daq-faux-<index>"""
-        self._run_shell('sudo cmd/faux %s' % index)
-        iface = 'faux-eth0'
-        prefix = int(index / 256) + 1
-        suffix = index % 256
-        ip_addr = '192.168.%s.%s' % (prefix, suffix)
-        gateway = '192.168.1.0'
-        container = 'daq-faux-%s' % index
-
     def create_veth_pair(self, iface1, iface2):
         """Creates a veth pair with interface names iface1, iface2"""
         self._run_shell('sudo ip link add dev %s type veth peer name %s' % (iface1, iface2))
         self._set_interface_up(iface1)
         self._set_interface_up(iface2)
+
+    def get_interface_ofport(self, iface):
+        """Returns ofport number of interface from ovs table"""
+        _, out, _ = self._run_shell('ovs-vsctl get Interface %s ofport' % iface)
+        return int(out.strip())
+
+    def get_forwarding_table(self, bridge):
+        """Returns forwarding table of given OVS bridge in the form [(<port>, <vlan>, <mac>)]"""
+        _, out, _ = self._run_shell('ovs-appctl fdb/show %s' % bridge)
+        return self._filter_forwarding_table(out)
+
+    def _filter_forwarding_table(self, table):
+        """Converts fdb table string to list[(<port>, <vlan>, <mac>)]"""
+        filtered_table = re.findall(r"\d{1,4}\s+\d{1,4}\s+(?:[0-9a-fA-F]:?){12}", table)
+        return [tuple(entry.split()) for entry in filtered_table]
+
+    def get_interface_ip(self, iface="eth0"):
+        """Returns IP of given interface"""
+        _, cmd_str, _ = self._run_shell('ip addr show %s' % iface)
+        return re.search(r'(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})', cmd_str).group(0)
```

Comment:
Should there be some error handling in case there is no interface ip?

Reply:
I think this should fail catastrophically. The burden of error handling should be on DAQ or those using these methods. I think this comes into play when device coupler is integrated with DAQ.

Reply:
I mean for error handling it can raise a sensible error here rather than “None type has no attribute group”.

Reply:
Makes sense.

Reply:
Changed to add error handling. I think rum_cmd should handle it well enough but this prevents dumping the whole stack trace.